openml_OS/views/pages/frontend/new/pre.php

```diff
@@ -33,7 +33,7 @@
 
 $this->datasets 			= $this->Dataset->getColumnWhere( 'name', 'isOriginal = "true"', '`name` ASC' );
 $this->datasetIds 			= $this->Dataset->getColumn( 'did', 'did' );
-$this->datasetVersion		= $this->Dataset->getColumnFunction( 'CONCAT(`name`,"(",`version`,")")', '`name` ASC' );
+$this->datasetVersion		= $this->Dataset->getColumnFunctionWhere( 'CONCAT(`name`,"(",`version`,")")', 'did IN (select did from dataset_status where status = "active")', '`name` ASC' );
```

Comment:
the `dataset_status` table holds a record and a date for every status change, and the latest status change is the actual one. Can you please reflect this in the query

Reply:
Sorry for missing this. I'll change it. I wasn't aware that was the semantic of the dataset_status table. So it actually means `data status updates`? FYI, currently there is only one entry for every dataset in table dataset_status. This returns an empty set: SELECT did FROM `dataset_status` group by did having count(did) > 1. Is that a coincidence? Also, since (did, status) is the primary key, does that mean you can never revert a status? Say that you change status from 'in_preparation' to 'active', can you revert it back to 'in_preparation'? Because the DB will complain that there is already an entry with key (data_id, 'in_preparation')

Reply:
IIUC, I need to find the datasets where the latest status update set the status to active, right. This is harder than I expected because MySQL doesn't support LIMIT inside IN subqueries. This is the most efficient way I could find. Is this correct?
```
select CONCAT(`name`,"(",`version`,")") from dataset d left outer join dataset_status ds on d.did = ds.did and status = 
(SELECT status FROM dataset_status where did=d.did ORDER BY status_date DESC LIMIT 1) where status='active'
```

Reply:
@janvanrijn any comments?